timemachine/fe/free_energy.py

```diff
@@ -1216,18 +1221,30 @@ def get_swap_acceptance_rates(fraction_accepted_by_pair):
             instantaneous_swap_acceptance_rates = get_swap_acceptance_rates(fraction_accepted_by_pair)
             average_swap_acceptance_rates = get_swap_acceptance_rates(np.sum(fraction_accepted_by_pair_by_iter, axis=0))
 
+            wall_time_per_frame_current = (current_time - last_update_time) / print_diagnostics_interval
+            wall_time_per_frame_average = (current_time - begin_loop_time) / (current_frame + 1)
+            estimated_wall_time_remaining = wall_time_per_frame_average * (md_params.n_frames - (current_frame + 1))
+
+            def format_wall_time(time_seconds):
+                return f"{time_seconds:.2f} s"
+
             def format_rate(r):
                 return f"{r * 100.0:5.1f}%"
 
             def format_rates(rs):
                 return " | ".join(format_rate(r) for r in rs)
 
-            print("Frame ", current_frame + 1)
-            print("Acceptance rates (inst.)   :", format_rates(instantaneous_swap_acceptance_rates))
-            print("Acceptance rates (average) :", format_rates(average_swap_acceptance_rates))
-            print("Final replica permutation  :", hrex.replica_idx_by_state)
+            print("Frame", current_frame + 1)
+            print("Wall time per frame, current   :", format_wall_time(wall_time_per_frame_current))
+            print("Wall time per frame, average   :", format_wall_time(wall_time_per_frame_average))
```

Comment:
Can we make these debug statements (logger.debug()) and/or expose control of this logging at a higher level? Right now, calling `run_vacuum` has no way to suppress these.

Reply:
I support the transition to logging, but would prefer to do this in a separate PR for all the various things that are currently logged via print statements (e.g. also minimization info, bisection iterations). I assume we'd want to keep the current level of logging in production, so maybe `logging.INFO` would be appropriate? For now, I suppose we could add a `verbose` flag to the `rbfe.run_x` functions, but even this might be better in a separate PR since it would change the status quo. What do you think?

Reply:
A separate PR is fine, and the changes as is are an overall improvement just want to be cognizant of the amount of logging we are putting out and the ability to control it at some higher level.